source/common/http/filter/ratelimit.cc

```diff
@@ -26,6 +26,19 @@ void ServiceToServiceAction::populateDescriptors(const Router::RouteEntry& route
       {{{"to_cluster", route.clusterName()}, {"from_cluster", config.localServiceCluster()}}});
 }
 
+void RequestHeadersAction::populateDescriptors(const Router::RouteEntry& route,
+                                               std::vector<::RateLimit::Descriptor>& descriptors,
+                                               FilterConfig&, const HeaderMap& headers) {
+  if (headers.has(header_name_)) {
+    if (!route.rateLimitPolicy().rateLimitKey().empty()) {
```

Comment:
We want both descriptors, not if/else. So always add the else one, then if there is a key that is not empty, add the other one.